pytoml/writer.py

```diff
@@ -1,6 +1,9 @@
 from __future__ import unicode_literals
 import io, datetime, math, sys
 
+if sys.version_info >= (3, 4):
```

Comment:
In this case it might be easier to detect features rather than versions.
```
try:
    from pathlib import PurePath as _path_types
except ImportError:
    _path_types = ()
```

Then later you can test for paths with `isinstance(v, _path_types)`.